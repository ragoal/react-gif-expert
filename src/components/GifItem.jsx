export const GifItem = ({title, url}) => {
    // return ( <img className="card" key={image.id} src={image.url} alt={image.title}/> );
    return ( 
        <div className="card">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    );
}
 