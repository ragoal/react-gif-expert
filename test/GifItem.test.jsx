import React from 'react';
import { render, screen } from '@testing-library/react';
import { GifItem } from '../src/components/GifItem';


describe('GifItem', () => {


    test('debe hacer match con snapshot', () => {

        const title = 'Test Title';
        const url = 'https://example.com/image.jpg';
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });


  test('renders the title and image correctly', () => {
    const title = 'Test Title';
    const url = 'https://example.com/image.jpg';

    const { getByAltText, getByText } = render(<GifItem title={title} url={url} />);
    const imageElement = getByAltText(title);
    const titleElement = getByText(title);

    expect(imageElement).toBeTruthy;
    expect(imageElement.src).toBe(url);
    expect(titleElement).toBeTruthy;
  });

  
});