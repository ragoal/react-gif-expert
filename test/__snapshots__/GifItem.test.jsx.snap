// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Title"
      src="https://example.com/image.jpg"
    />
    <p>
      Test Title
    </p>
  </div>
</div>
`;
